### PR TITLE
Modify to use latest version of Nexmo and client library

### PR DESCRIPTION
**Pull request by marklewin**
Updated the tutorial:

- Changed the current `nexmo` dependency to 5.4.0
- Changed the calls to the insight API to reflect the new methods (e.g. `number_insight.basic()` instead of `get_basic_number_insight()`)
- Changed the pricing API example (you don't seem to be able to pass a number in as a parameter any more, so I'm using the standard instead of basic NI API to retrieve both the `country_code` and `network_code` and using them to calculate pricing)
- Switched the ordering of the `cost` and `channels` snippets to achieve a better flow in terms of building on previous examples
- Updated the `README`

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
 source 'https://rubygems.org/'
 
-gem 'nexmo', '~> 4.4.0'
+gem 'nexmo', '~> 5.4.0'
```
Comment (TheChrisVela) [inline Gemfile:3]: Nice catch

File: snippets/4_cost.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+current_network = insight.current_carrier.network_code
+current_country = insight.country_code
+
+# Fetch the voice and SMS pricing data for the country
+sms_pricing = nexmo.pricing.sms.get(current_country)
+voice_pricing = nexmo.pricing.voice.get(current_country)
+
+# Retrieve the network cost from the pricing data
+sms_cost = sms_pricing.networks.select{
+  |network| network.network_code == current_network}
```
Comment (TheChrisVela) [inline snippets/4_cost.rb]: I usually put `|network|` on the same line as `.select{`

File: snippets/4_cost.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+voice_cost = voice_pricing.networks.select{
+  |network| network.network_code == current_network}
```
Comment (TheChrisVela) [inline snippets/4_cost.rb]: I usually put `|network|` on the same line as `.select{`

File: snippets/4_cost.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+p({
```
Comment (TheChrisVela) [inline snippets/4_cost.rb:36]: Oh this is much better than `put foo.inspect` nice!

**Review by TheChrisVela — CHANGES_REQUESTED** [review]
Minor styling comment. If you disagree we can skip and I'll approve and merge.

File: Gemfile (hunk 1 above)
Comment (marklewin) [reply Gemfile:3]: Thanks!

File: snippets/4_cost.rb (hunk 2 above)
Comment (marklewin) [reply snippets/4_cost.rb]: Done!

File: snippets/4_cost.rb (hunk 3 above)
Comment (marklewin) [reply snippets/4_cost.rb]: Done!

File: snippets/4_cost.rb (hunk 4 above)
Comment (marklewin) [reply snippets/4_cost.rb:36]: Thanks.

**Comment by marklewin** [comment]
Thanks for the review and kind comments. I have made the required styling changes to `/snippets/4_cost.rb`.

**Comment by marklewin** [comment]
Closed in error, reopening.

**Review by TheChrisVela — APPROVED** [review]

**Pull request merged**